### DISCUSSION TITLE
Adding rich content to Markers

### DISCUSSION THREAD

**Discussion by jaybo** [jaybo/deepzoom-blog#3] [Ideas]
From the outset of DeepZoom V2, I've been contemplating whether to add user generated rich content (text, images, video) to Markers.  

There are lots of websites which attach commercial content to map markers showing marinas, gas prices, and the like, but I've been imagining something closes to a mashup of Twitter and Wikipedia, where you can add comments on anchorages, dive spots, fishing holes, kayaking campgrounds, etc. along with images, videos, and ratings or likes/dislikes. 

But entering this arena means higher operating costs and the more serious problem of moderating user submissions.  

If you have any ideas or suggestions on this topic, I'd love to hear them.

**Comment by jaybo** [comment]
Rich content for markers and trips was added in version 2.7.

**Comment by Cmjstealth** [comment]
You still planning on releasing functionality for promotion of private markers to public? As a frequent user of deepzoom I'd be happy to help input some popular salish sea dive sites. Unless someone already has done this work in which case I'll probably only create markers intended for my own use. If just for myself I wouldn't go through the hassle of adding the same level of details, pictures, and won't bother creating markers for sites I know so well I have memorized. 

It also appears like personal markers are always visible on the map while public markers are only visible at higher zoom levels. Is this true? In this case to avoid clutter I certainly don't want to create markers I don't need personally if they arn't of value to others.

**Reply by jaybo** [reply]
Yes, this is still definitely part of the game plan.  I've been working on other aspects of DeepZoom for the last few months, but it's time to revisit this topic.  

Private markers that become public will have the same zoom visibility properties as all other public markers, so there's no issue with increasing clutter.

I haven't settled on how to expose this, but here's one possibility:

- Have a "Make public" checkbox on the Trips page.
- The "demo" page becomes a list of filterable categories, maybe something like:

1. demos
2. highly ranked trips
3. recently added trips and markers
4. vote on public markers
5. blog posts

- This would allow public markers to be visible and allow them to be upvoted.

But I'm certainly open to other alternatives.  

At any rate, I'd love to just verify the code to convert private markers to public is actually working, so if you have a trip with a marker or two I can experiment with I'd like to test it out.

**Comment by jaybo** [comment]
There is now a dialog to convert your private markers into public markers.
www.deepzoom.com/add-public-markers
